src/wallet.ts

```diff
@@ -590,14 +600,37 @@ async function esploraTxToTxInterface(
   );
 
   const txHex = await fetchTxHex(esploraTx.txid, explorerUrl);
-  const txOutputs = Transaction.fromHex(txHex).outs.map(
-    txOutputToOutputInterface
-  );
+  const transaction = Transaction.fromHex(txHex);
+
+  const txOutputs = transaction.outs.map(txOutputToOutputInterface);
+
+  const feeOutput = transaction.outs
+    // fee output is never blinded
+    .filter((out: TxOutput) => !isConfidentialOutput(out))
+    .find((out: TxOutput) => out.script.toString('hex').valueOf() === '');
+
+  if (!feeOutput) throw new Error('Fee output is not found.');
 
   const tx: TxInterface = {
     txid: esploraTx.txid,
     vin: txInputs,
     vout: txOutputs,
+    fee: {
+      amount: toNumber(feeOutput.value),
+      asset: toAssetHash(feeOutput.asset),
+    },
+    status: {
+      confirmed: esploraTx.status.confirmed,
+      blockHash: esploraTx.status.confirmed
```

Comment:
```suggestion
      blockHash: esploraTx.status.confirmed && esploraTx.status.block_hash
```

This should already return undefined if not true the condition